python/tank/deploy/tank_commands/get_entity_commands.py

```diff
@@ -0,0 +1,282 @@
+from ...errors import TankError
+from ..util import execute_toolkit_command, SubprocessCalledProcessError
+
+import itertools
+import operator
+import os
+import sys
+
+
+class GetEntityCommandsAction(Action):
+    """
+    Gets the commands that can be launched on certain entities for another
+    pipeline configuration.
+
+    This is done by calling the tank command on the other pipeline
+    configuration and asking for its cached entity commands (or ask to update
+    its cache beforehand if needed).
+
+    It is used like this:
+    >>> import tank
+    # create our command object
+    >>> cmd = tank.get_command("get_entity_commands")
+    # get the commands for tasks, but could mix and match with any other types
+    >>> tasks = [("Task", 1234), ("Task", 1235)]
+    >>> commands_by_task = cmd.execute({"pc_path": "/my/pc/path",
+    >>>                                 "entities": tasks})
+    # extract the commands of a specific task
+    >>> commands = commands_by_task[tasks[0]]
+    """
+    def __init__(self):
+        Action.__init__(self,
+                        "get_entity_commands",
+                        Action.GLOBAL,
+                        ("Gets the available commands that can be executed "
+                         "for specified entities from another pipeline "
+                         "configuration"),
+                        "API")
+
+        # no tank command support for this one because it returns an object
+        self.supports_tank_command = False
+
+        # this method can be executed via the API
+        self.supports_api = True
+
+        self.parameters = {
+            "pc_path": {
+                "description": "Path to the pipeline configuration associated "
+                               "with the entities.",
+                "type":        "str"
+            },
+
+            "entities": {
+                "description": """List of entities to fetch the actions for.
+                                  Every entity should be a tuple with the
+                                  following format:
+                                    (entity_type, entity_id)""",
+                "type":        "list"
+            },
+
+            "return_value": {
+                "description": """Dictionary of the commands by entity, with
+                                  the (entity_type, entity_id) tuple used as a
+                                  key. Each value is a list of commands. A
+                                  command is a dictionary with the following
+                                  format:
+                                    {
+                                      "name":  command to execute
+                                      "title": title to display for the command
+                                      "icon":  path to the icon of this command
+                                    }""",
+                "type":        "dict"
+            }
+        }
+
+    _ERROR_CODE_CACHE_OUT_OF_DATE = 1
```

Comment:
I think these constants should be placed right after the class docstring. That's typically how we do it - that way you don't need to hunt for constants in the file.